### ISSUE TITLE
Conflict between AsyncCohortDeletion and AsyncEventDeletion causing assert exception

### ISSUE THREAD

**Issue by fazlali** [PostHog/posthog#15014]
# Description:
Two asynchronous processes `AsyncCohortDeletion` and `AsyncEventDeletion` conflict with each other, resulting in an assert exception being raised at `posthog/models/async_deletion/delete_events.py:86`.

The root cause of this issue is that the parent class (`AsyncDeletionProcess`) passes all unhandled deletions to the child `process` method, which leads to `AsyncEventDeletion` and `AsyncCohortDeletion` both attempting to handle deletions that they are not designed to handle. As a result, the assert exception is triggered at `posthog/models/async_deletion/delete_events.py:86`.

# Possible solution:
A possible solution to resolve this conflict is to implement a filtering mechanism in the process methods of both `AsyncEventDeletion` and `AsyncCohortDeletion` classes, to ensure that only the relevant deletions are processed. For `AsyncEventDeletion`, the deletions should be filtered to include only `DeletionType.Person`, `DeletionType.Group`, and `DeletionType.Team`, as demonstrated in the following example:

```
class AsyncEventDeletion(AsyncDeletionProcess):
    def process(self, deletions: List[AsyncDeletion]):
        deletions = [
            deletion
            for deletion in deletions
            if deletion.deletion_type in (
                DeletionType.Person,
                DeletionType.Group,
                DeletionType.Team,
            )
        ]
        # Perform other tasks...
```

Similarly, for `AsyncCohortDeletion`, the deletions should be filtered to include only `DeletionType.Cohort_full` and `DeletionType.Cohort_stale`, as shown in the following example:

```
class AsyncCohortDeletion(AsyncDeletionProcess):
    def process(self, deletions: List[AsyncDeletion]):
        deletions = [
            deletion
            for deletion in deletions
            if deletion.deletion_type in (
                DeletionType.Cohort_full,
                DeletionType.Cohort_stale,
            )
        ]
        # Perform other tasks...
```
By implementing this solution, the conflict between `AsyncCohortDeletion` and `AsyncEventDeletion` can be resolved, and the assert exception can be prevented from occurring.

**Comment by logand22** [comment]
Was this issue fixed by https://github.com/PostHog/posthog/pull/15019?

**Comment by fazlali** [comment]
Yes, the issue was resolved with pull request #15019
